File: Python/Binary Tree to DLL.py
class Solution:
    def bToDLL(self,root):
        # do Code her
        self.head = self.prev = None
        self.dfs(root)
        return self.head
        
    def dfs(self,root):
        if root == None:
            return 
        
        self.dfs(root.left)
        if self.head == None:
            self.head = root
            
        root.left = self.prev
        if self.prev == None:
            self.prev = root
        else:
            self.prev.right = root
            self.prev = root
            
            
        self.dfs(root.right)
        return

File: Java/Binary Tree to DLL.java
class Solution
{
    Node head = null, prev = null;
    //Function to convert binary tree to doubly linked list and return it.
    Node bToDLL(Node root)
    {
	//  Your code here	
	dfs(root);
	return head;
    }
    void dfs(Node root){
        if(root==null)return;
        dfs(root.left);
        if(head==null){
            head = root;
        }
        root.left = prev;
        if(prev==null){
            prev = root;
        }else{
            prev.right = root;
            prev = root;
        }
        dfs(root.right);
        return;
    }
}